### PR TITLE
XD-3492 - Add Header-Enricher

### PR DESCRIPTION
**Pull request by dturanski**

### CODE REVIEW

File: extensions/spring-xd-extension-header-enricher/src/test/java/org/springframework/xd/header/enricher/HeaderEnricherIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	 * Start the single node container, binding random unused ports, etc. to not conflict with any other instances
+	 * running on this host. Configure the ModuleRegistry to include the project module.
+	 */
+	@BeforeClass
+	public static void setUp() {
+		System.setProperty("XD_HOME", "../..");
+		RandomConfigurationSupport randomConfigSupport = new RandomConfigurationSupport();
+		application = new SingleNodeApplication().run();
+		SingleNodeIntegrationTestSupport singleNodeIntegrationTestSupport = new SingleNodeIntegrationTestSupport(
+				application);
```
Comment (garyrussell) [inline extensions/spring-xd-extension-header-enricher/src/test/java/org/springframework/xd/header/enricher/HeaderEnricherIntegrationTest.java]: These variables are not used (randomConfigSupport and singleNodeIntegrationTestSupport) and the tests run ok - is there some magic they provide that I am not aware of?

Comment (dturanski) [inline extensions/spring-xd-extension-header-enricher/src/test/java/org/springframework/xd/header/enricher/HeaderEnricherIntegrationTest.java]: they can be removed

Comment (garyrussell) [inline extensions/spring-xd-extension-header-enricher/src/test/java/org/springframework/xd/header/enricher/HeaderEnricherIntegrationTest.java]: Confused - the comment says it starts the app on random ports - how does that happen without RandomConfigurationSupport.

Also, don't we need an AfterClass with `application.close()` ??

**Comment by garyrussell** [comment]
Merged as 0966eb3ae5f0218bb4d06a59e3b64195419cf272 after renaming tests and stopping the singlenode app in `@AfterClass`.

**Pull request closed without merging**